JavaScript1/Week2/homework.md

```diff
@@ -6,200 +6,157 @@
+---
 
+1. Create a function that returns an array with with 3 empty objects.
+2. Call the function and assign the return of the function to a variable called colorObjects (like we did in the previous exercise)
+3. Change the array so that all the objects have one key called color. Make every object have a different color
+4. Log out colorObjects
+5. Give the function 3 parameters called color1, color2 and color3. The parameters should control the color of the 3 objects. color1 sets the color of the first object and so on. 
+6. Call the function with 3 colors and log out colorObjects. colorObjects should look like this:
 ```js
-var obj1 = {
-  a: 1,
-  b: "this is the letter b",
-  c: {
-    foo: "what is a foo anyway",
-    bar: [1, 2, 3, 4]
-  }
-};
-
-var obj2 = {
-  a: "1",
-  b: "this is the letter b",
-  c: {
-    foo: "what is a foo anyway",
-    bar: [1, 2, 3, 4]
-  }
-};
+[{
+    color: 'red',
+},{
+    color: 'blue',
+},{
+    color: 'yellow',
+}]
 ```
+If the colors you called the function with were red, blue and yellow.
 
-    In our example we'll say that `obj1 == obj2` is `true` and `obj1 === obj2` is `false`. Make sure you can see why before you write any code!
-    
-    Note: give this exercise your best shot but don’t spend more than, say, one hour on it.
+7. For every object in the array log out the color of the object. Hint use a for loop and read [Here](https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/objects.md) to check how to get the value of an object.
 
-14. We saw that we can pass functions as arguments to other functions. Your task is to write a function that takes another function as an argument and runs it. 
+---
 
-```js
-    function foo(func) {
-        // What to do here? 
-    }
-    
-    function bar() {
-        console.log('Hello, I am bar!');
-    }
-    
-    foo(bar);
-```
+8. Today is Sunday and you ask your friend in how many days you are meeting. If the friend says
+I will see you in 9 days. The output should be Tuesday. You should get the today's day from the system.
+So typical console output is:
 
+``Today is: Sunday``    
+``How many days to meet : 9``  
+``We are meeting on : Tuesday``  
 
-15. Write some code to test two arrays for equality using `==` and `===`. Test the following:
-    
-```js
-var x = [1, 2, 3];
-var y = [1, 2, 3];
-var z = y;
-```
+Hint: use remainder operator
 
-What do you think will happen with `x == y`, `x === y` and `z == y` and `z == x`? Prove it!
-    
-> Don't cheat! Seriously - try it first.
-    
+## Step 3: Smart-ease - Goes Global!
+Following the massive success of Smart-ease's first couple of products things have been quite hectic at Smart-ease's shared office space. The phone is constantly ringing from journalists wanting to interview the genius behind the success and the greatest developers want to work for you. Wired wrote an article with the headline "Smart-ease as good as Smarties"
```

Comment:
Could add a link to Wired.com, just so people understand it as a well-known technology news outlet.